packages/discovery-provider/plugins/pedalboard/apps/solana-relay/src/routes/relay/assertRelayAllowedInstructions.ts

```diff
@@ -324,6 +323,24 @@ const assertAllowedSystemProgramInstruction = (
   }
 }
 
+const assertValidSecp256k1ProgramInstruction = (
+  instructionIndex: number,
+  instruction: TransactionInstruction
+) => {
+  try {
+    if (
+      !Secp256k1Program.verifySignature(Secp256k1Program.decode(instruction))
+    ) {
+      throw new Error('Signer does not match')
+    }
+  } catch (e) {
```

Comment:
nit: This catch block will hide unexpected errors and resurface them as if the instruction is invalid.
It would be nice if we could use specific errors classes to indicate failures we know of, and then catch other errors and resurface them in a way that makes it clear it's an unexpected error.